onadata/libs/permissions.py

```diff
@@ -300,47 +347,57 @@ class OwnerRole(Role):
 # Memoize a class to permissions dict.
 for role in ROLES.values():
     role.class_to_permissions = defaultdict(list)
-    [role.class_to_permissions[k].append(p) for p, k in role.permissions]
+    for p, k in role.permissions:
```

Comment:
This is to not create another object? This may have been written the way it was because we thought list comprehension was faster, did you profile this change? Definitely the comprehension would mean memory allocation (at least briefly until the function got returned, but I suppose even it that case it might not be released but merely marked for GC)

Reply:
The main reason for this change had to do with Pylint complaining, the comprehension needed to be assigned to something.

```vim
pylint: [expression-not-assigned] Expression "[role.class_to_permissions[k].append(p) for (p, k) in role.permissions]" is assigned to nothing [W0106]
```
I will revisit and see if it can be done differently.